### DISCUSSION TITLE
Entropy of the water does not validate with the literature at room temperature.

### DISCUSSION THREAD

**Discussion by armahapa** [reaktoro/reaktoro#406] [General]
I was trying to use the supcrt98 database for entropy calculation for an electrolyte Solution. Just to validate the results for water and calculated entropy difference at two different temperatures with the following code:

from reaktoro import *
db = SupcrtDatabase("supcrt98")
system = ChemicalSystem(db,
    AqueousPhase("Na+ Cl- H2O(aq)").set(ActivityModelDebyeHuckel()),
)
dT=[10, 1, 0.1, 0.001, 1e-4,1e-5, 1e-9]
dP=[1000,500, 100, 10, 1, 0.1, 0.001, 1e-4,1e-5, 1e-9]
state = ChemicalState(system)
state.temperature(300.0, "K")
state.pressure(1, "bar")
state.set("H2O(aq)", 1.0, "kg")
state.set("Na+", 1e-18, "mol")
state.set("Cl-", 1e-18, "mol")
solver = SmartEquilibriumSolver(system)
state.props().update(state)
props=state.props()
h=props.specificEnthalpy()
s=props.specificEntropy()
G=props.specificGibbsEnergy()
s0=55*props.speciesStandardEntropy('H2O(aq)')

state.temperature(325.0, "K")
state.props().update(state)
props=state.props()
print('enthalpy difference')
print(props.specificEnthalpy()-h)
print('entropy difference')
print(props.specificEntropy()-s)
print('entropy difference of the standard state')
print(55*props.speciesStandardEntropy('H2O(aq)')-s0)

Here is the output:
enthalpy difference
104500
entropy difference
1323.74
entropy difference of the standard state
1311.61

I tried to validate the value from NIST (https://webbook.nist.gov/cgi/fluid.cgi?P=1&TLow=300&THigh=350&TInc=5&Digits=5&ID=C7732185&Action=Load&Type=IsoBar&TUnit=K&PUnit=bar&DUnit=kg%2Fm3&HUnit=kJ%2Fkg&WUnit=m%2Fs&VisUnit=uPa*s&STUnit=N%2Fm&RefState=DEF)
the enthalpy difference matches with reaktoro, but the entropy difference comes as: 334.58 J/kg-K.

Interestingly the ratio is around 4. I was wondering if this difference occurs from the conversion from cal to J, or mol to mass of water. supcrt database happens to use cal more frequently in their database. Or, we are missing something.

Thanks in advance.

**Comment by allanleal** [comment]
Hi @armahapa - thanks for initiating this discussion.

There are two things to pay attention to here.

1. The HKF model assumes the triple point of water as the reference state
2. Reaktoro at the moment is computing the standard-state entropy (S0) of a species from its standard-state Gibbs energy and enthalpy: S0 = (H0 - G0)/T

The script below demonstrates that once these points are observed, the values for entropy are consistent to that of NIST and the Wagner and Pruss equation of state:

```py
from reaktoro import *

db = SupcrtDatabase("supcrt98")

system = ChemicalSystem(db, 
    AqueousPhase("Na+ Cl- H2O(aq)").set(ActivityModelDebyeHuckel())
)

T0 = 300.0
T1 = 325.0

state0 = ChemicalState(system)
state0.temperature(T0, "K")
state0.pressure(1, "bar")
state0.props().update(state0)
props0 = state0.props()

state1 = ChemicalState(system)
state1.temperature(T1, "K")
state1.pressure(1, "bar")
state1.props().update(state1)
props1 = state1.props()

S0_HKF = 0.055508 * props0.speciesStandardEntropy('H2O(aq)')  # from J/mol/K to J/g/K
S1_HKF = 0.055508 * props1.speciesStandardEntropy('H2O(aq)')  # from J/mol/K to J/g/K

S0_WP = 0.001 * waterThermoPropsWagnerPruss(300, 1e5, StateOfMatter.Liquid).S.val()  # from J/kg/K to J/g/K
S1_WP = 0.001 * waterThermoPropsWagnerPruss(325, 1e5, StateOfMatter.Liquid).S.val()  # from J/kg/K to J/g/K

Ttr = 273.16  # the triple point temperature of water in K
Str = 63.312288  # the standard entropy of water at 298.15 K in J/mol/K
Gtr = -235517.36  # the standard Gibbs free energy of water at 298.15 K in J/mol
Htr = -287721.13  # the standard enthalpy of water at 298.15 K in J/mol

S0_HKF_correction = ((Htr - Gtr)/T0 + Str - Str*Ttr/T0) * 0.055508
S1_HKF_correction = ((Htr - Gtr)/T1 + Str - Str*Ttr/T1) * 0.055508

S0_HKF_corrected = S0_HKF - S0_HKF_correction
S1_HKF_corrected = S1_HKF - S1_HKF_correction

print(f"S0_HKF: {S0_HKF:.4f} J/g/K")
print(f"S1_HKF: {S1_HKF:.4f} J/g/K")
print(f"S1_HKF - S0_HKF: {S1_HKF - S0_HKF:.4f} J/g/K")
print()
print(f"S0_WP: {S0_WP:.4f} J/g/K")
print(f"S1_WP: {S1_WP:.4f} J/g/K")
print(f"S1_WP - S0_WP: {S1_WP - S0_WP:.4f} J/g/K")
print()
print(f"S0_HKF_correction: {S0_HKF_correction:.4f} J/g/K")
print(f"S1_HKF_correction: {S1_HKF_correction:.4f} J/g/K")
print(f"S1_HKF_correction - S0_HKF_correction: {S1_HKF_correction - S0_HKF_correction:.4f} J/g/K")
print()
print(f"S0_HKF_corrected: {S0_HKF_corrected:.4f} J/g/K")
print(f"S1_HKF_corrected: {S1_HKF_corrected:.4f} J/g/K")
print(f"S1_HKF_corrected - S0_HKF_corrected: {S1_HKF_corrected - S0_HKF_corrected:.4f} J/g/K")
```

**Output**
```text
S0_HKF: -8.9516 J/g/K
S1_HKF: -7.6279 J/g/K
S1_HKF - S0_HKF: 1.3237 J/g/K

S0_WP: 0.3931 J/g/K
S1_WP: 0.7276 J/g/K
S1_WP - S0_WP: 0.3346 J/g/K

S0_HKF_correction: -9.3447 J/g/K
S1_HKF_correction: -8.3555 J/g/K
S1_HKF_correction - S0_HKF_correction: 0.9892 J/g/K

S0_HKF_corrected: 0.3931 J/g/K
S1_HKF_corrected: 0.7276 J/g/K
S1_HKF_corrected - S0_HKF_corrected: 0.3346 J/g/K
```

You may want to see how the standard-state properties of water are computed when using HKF model (via SUPCRT databases):

https://github.com/reaktoro/reaktoro/blob/f8416a75b295231ae98e4d9fd0ebc03b4c0337ab/Reaktoro/Models/StandardThermoModels/StandardThermoModelWaterHKF.cpp#L29

Let me know if this makes sense.

**Reply by armahapa** [reply]
Thank you, this transformation makes sense. I tried this transformation on different reference points, and the values are now consistent.
